### PR TITLE
Add installation steps for Podman 5 in CI workflows

### PR DESCRIPTION
**Pull request by ericcurtin**
This ensures that Podman 5 is available for use in CI jobs.

## Summary by Sourcery

CI:
- Add installation steps for Podman 5 in CI workflows across multiple YAML configuration files.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,6 +19,12 @@ jobs:
            make install-requirements
            pip install tqdm --break-system-packages
 
+      - name: Install podman 5
+        run: |
+           set -e
+           sudo curl -OL http://archive.ubuntu.com/ubuntu/pool/universe/libp/libpod/podman_5.0.3+ds1-5ubuntu1_amd64.deb
+           sudo dpkg -i --force-depends --force-overwrite podman_5.0.3+ds1-5ubuntu1_amd64.deb
+
```
[the remark this answers is not in the document]
Comment (ericcurtin) [reply .github/workflows/ci.yml]: Im gonna try this next and see if it works, if it works, its better than hardcoding .deb files

**Comment by ericcurtin** [comment]
@rhatdan if this passes, it should be good and bring us podman 5

**Comment by ericcurtin** [comment]
Has potential to break things as other packages might start to get installed from there, it is a bleeding edge repo

**Comment by ericcurtin** [comment]
FrankenUbuntu 24.04/24.10 is working and we now have podman 5 in CI:

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**